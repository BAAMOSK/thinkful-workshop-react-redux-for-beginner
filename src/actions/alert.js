export function closeAlert() {
  return {
    type: 'CLOSE_ALERT',
  };
}
