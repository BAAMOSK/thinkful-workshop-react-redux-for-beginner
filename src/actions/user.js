import { push } from 'react-router-redux';
import { toastr } from 'react-redux-toastr';

export function login(values) {
  return (dispatch) => {
    if (values.email === 'quimperemanuel@gmail.com' && values.password === '123') {
      dispatch({
        type: 'LOGIN',
      });

      dispatch(push('/'));

      return dispatch(toastr.success('Welcome', values.email));
    }

    return dispatch({ type: 'OPEN_ALERT', alertType: 'UNAUTHORIZED' });
  };
}

export function logout() {
  return {
    type: 'LOGOUT',
  };
}
