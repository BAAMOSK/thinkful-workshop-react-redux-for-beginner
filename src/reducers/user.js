const initialState = {
  isLogged: true,
  isAdmin: false,
};

export default (state = initialState, action) => {
  switch (action.type) {
    default:
      return state;
  }
};
