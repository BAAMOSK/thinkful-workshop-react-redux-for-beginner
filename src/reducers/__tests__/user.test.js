import userReducer, { initialState } from '../user';
import { login, logout } from '../../actions/user';

describe('user: reducer', () => {
  describe('#login()', () => {
    it('should log a user', () => {
      expect(
        userReducer(
          initialState,
          login({
            email: 'quimperemanuel@gmail.com',
            password: '123',
          })
        )
      ).toMatchSnapshot();
    });
  });
  describe('#logout()', () => {
    it('should logout a user', () => {
      expect(
        userReducer(
          initialState,
          logout()
        )
      ).toMatchSnapshot();
    });
  });
});
