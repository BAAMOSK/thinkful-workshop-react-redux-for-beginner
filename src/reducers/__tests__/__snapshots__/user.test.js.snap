exports[`user: reducer #login() should log a user 1`] = `
Object {
  "isAdmin": false,
  "isLogged": true,
}
`;

exports[`user: reducer #logout() should logout a user 1`] = `
Object {
  "isAdmin": false,
  "isLogged": false,
}
`;
